### PR TITLE
adds optional case-file parameter to the online itool command

### PR DESCRIPTION
**Pull request by CBiasuzzi**

### CODE REVIEW

**Comment by mathbagu** [comment]
I'm not sure this is compliant to the need: we should not be able to mix the options (case-date, country, case-type) and (case-file). These two parameter groups are exclusive. I think, if an user mix the options, itools command should display the usage message.

File: modules/src/main/java/eu/itesla_project/modules/online/OnlineWorkflowParameters.java [hunk 1]
```diff
@@ -114,6 +117,7 @@ public OnlineWorkflowParameters(DateTime baseCaseDate, int states, Interval hist
         this.limitReduction = limitReduction;
         this.handleViolationsInN = handleViolationsInN;
         this.constraintMargin = constraintMargin;
+        this.caseFile = caseFile;
```
Comment (mathbagu) [inline modules/src/main/java/eu/itesla_project/modules/online/OnlineWorkflowParameters.java]: The input parameters check should be more complex: (case-date, countries, case-type) are all not null xor case-file is not null.

File: online-workflow/src/main/java/eu/itesla_project/online/OnlineWorkflowImpl.java [hunk 2]
```diff
@@ -164,8 +169,16 @@ public void start(OnlineWorkflowContext oCtx) throws Exception {
...
-                loadFlowFactory, 0, mergeOptimizerFactory, computationManager, parameters.isMergeOptimized());
+        Network network = null;
+        if (parameters.getCaseFile() != null ) {
+            network = Importers.loadNetwork(parameters.getCaseFile());
+            if (network == null) {
+                throw new RuntimeException("Case '" + parameters.getCaseFile() + "' not found");
+            }
+        } else {
+            network = MergeUtil.merge(caseRepository, parameters.getBaseCaseDate(), parameters.getCaseType(), parameters.getCountries(),
+                    loadFlowFactory, 0, mergeOptimizerFactory, computationManager, parameters.isMergeOptimized());
```
Comment (mathbagu) [inline online-workflow/src/main/java/eu/itesla_project/online/OnlineWorkflowImpl.java]: Why OnlineWorkflow depend on MergeUtil? Maybe using the caseRepository is sufficient? Is this dependency really needed?

File: online-workflow/src/main/java/eu/itesla_project/online/tools/OnlineWorkflowCommand.java [hunk 3]
```diff
@@ -43,6 +43,7 @@
     public static final String LIMIT_REDUCTION = "limits-reduction";
     public static final String HANDLE_VIOLATION_IN_N = "handle-violations";
     public static final String CONSTRAINT_MARGIN = "constraint-margin";
+    public static final String CASE_FILE="case-file";
```
Comment (mathbagu) [inline online-workflow/src/main/java/eu/itesla_project/online/tools/OnlineWorkflowCommand.java]: add spaces before and after =

File: online-workflow/src/main/java/eu/itesla_project/online/tools/OnlineWorkflowTool.java [hunk 4]
```diff
@@ -95,6 +95,9 @@ public void run(CommandLine line) throws Exception {
             }
         }
 
+        if(line.hasOption(OnlineWorkflowCommand.CASE_FILE))
```
Comment (mathbagu) [inline online-workflow/src/main/java/eu/itesla_project/online/tools/OnlineWorkflowTool.java]: Add space between "if" and "("

File: modules/src/main/java/eu/itesla_project/modules/online/OnlineWorkflowParameters.java (hunk 1 above)
Comment (CBiasuzzi) [inline modules/src/main/java/eu/itesla_project/modules/online/OnlineWorkflowParameters.java]: it's true that it's not enforced, but the command help explains how's working :
"case file - Note:  override (base-case, case-type, countries) parameters"
Anyway, I'll add a more "foolproof" check, if you deem it necessary ...

File: online-workflow/src/main/java/eu/itesla_project/online/OnlineWorkflowImpl.java (hunk 2 above)
Comment (CBiasuzzi) [inline online-workflow/src/main/java/eu/itesla_project/online/OnlineWorkflowImpl.java]: the WF could  possibly work with more than one country (... and merge was used, during the iTesla project)

Comment (mathbagu) [inline online-workflow/src/main/java/eu/itesla_project/online/OnlineWorkflowImpl.java]: Fine

File: modules/src/main/java/eu/itesla_project/modules/online/OnlineWorkflowParameters.java (hunk 1 above)
Comment (mathbagu) [inline modules/src/main/java/eu/itesla_project/modules/online/OnlineWorkflowParameters.java]: Could you try to implement an exclusive choice between the two modes: (case-date, country, case-type) ou (case-file)?

**Pull request merged**